### PR TITLE
Add Cluster filter on Kubernetes Overview ECS dashboard

### PR DESCRIPTION
**Pull request by akshay-saraswat**
## What does this PR do?

Added a Cluster name selector on the `[Metricbeat Kubernetes] Overview ECS` dashboard. This selector will be used as a filter for the dashboard. Also, added a blur for this selector on the Kubernetes module documentation page to explain how this selector works and where it sources the data from. Added a screenshot of the `[Metricbeat Kubernetes] Overview ECS` dashboard on the Kubernetes module documentation page because we need a visual to explain the Cluster filter.

## Why is it important?

We have added a new Cluster name field in metricbeat. this field will not be useful unless users could discover it in some form 
in UI.

## Checklist

- [x] I have made corresponding changes to the documentation
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Screenshots

Screen Shot Kubernetes Dashboard

### CODE REVIEW

**Review by ChrsMark — COMMENTED** [review]
Overall looks good to me, thanks for working on this @akshay-saraswat  but I think the Dashboard screenshot needs to be replaced with a more representative one displaying nodes, deployments, etc as well as the usage of the cluster filter.

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -998,6 +998,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add total CPU to vSphere virtual machine metrics. {pull}26167[26167]
 - Add AWS Kinesis metricset. {pull}25989[25989]
 - Move openmetrics module to oss. {pull}26561[26561]
+- Add Cluster filter on ECS Kubernetes overview dashboard and corresponding section on Kubernetes module documentation page. {pull}[26919]
```
Comment (MichaelKatsoulis) [inline CHANGELOG.next.asciidoc:1001]:
```suggestion
- Add Cluster filter on ECS Kubernetes overview dashboard and corresponding section on Kubernetes module documentation page. {pull}26919[26919]
```

**Review by ChrsMark — APPROVED** [review]
lgtm!

Feel free to merge it once you resolve https://github.com/elastic/beats/pull/26919#discussion_r671067948 and CI is green (I think you need to sign the CLA).

File: CHANGELOG.next.asciidoc (hunk 1 above)
Comment (akshay-saraswat) [reply CHANGELOG.next.asciidoc]: Added.

**Review by kaiyan-sheng — APPROVED** [review]

**Comment by jsoriano** [comment]
Dashboard added here has some kind of issue, `test_dashboards` fails with:
```
E         Exiting: Failed to import dashboard: Failed to load directory /go/src/github.com/elastic/beats/metricbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/7/dashboard:
E           error loading /go/src/github.com/elastic/beats/metricbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/7/dashboard/Metricbeat-kubernetes-overview.json: returned 500 to import file: <nil>. Response: {"statusCode":500,"error":"Internal Server Error","message":"An internal server error occurred."}
E         
E       assert 1 == 0
```
`test_dashboards` can be executed with `PYTEST_ADDOPTS="-k test_dashboards" mage pythonIntegTest` from the `metricbeat` directory.

It may be because it uses version 7.14.0 that hasn't been released yet, but not sure.

Why wasn't this detected by CI? These were the logs [there](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats/detail/PR-26919/5/pipeline/2065):
```
[2021-07-16T20:05:02.815Z] >> python test: Integration-kubernetes Testing
[2021-07-16T20:05:02.815Z] Cache entry deserialization failed, entry ignored
[2021-07-16T20:05:20.912Z] >> python test: Integration-kubernetes Testing - No tests found.
```
It seems that selective testing was triggered as only one module was modified, and no test was executed in this case. `test_dashboards` should have been run to detect this issue.

@ChrsMark @akshay-saraswat could you please review this dashboard?

cc @sayden @kvch

**Comment by ChrsMark** [comment]
@jsoriano I think that Dashboard is ok. The backports to 7.x and 7.14 should work with [7.14 compatible Dasboards](https://github.com/elastic/beats/pull/26919/files#diff-30efb8bddf5c54202fd98ff32cd8385fe81163254f48b64f0e3f73bbac603430R37). I think that we miss the update of the testing stack, sth like https://github.com/elastic/beats/pull/24957.

**Comment by jsoriano** [comment]
> @jsoriano I think that Dashboard is ok. The backports to 7.x and 7.14 should work with [7.14 compatible Dasboards](https://github.com/elastic/beats/pull/26919/files#diff-30efb8bddf5c54202fd98ff32cd8385fe81163254f48b64f0e3f73bbac603430R37). I think that we miss the update of the testing stack, sth like #24957.

But `test_dashboards` is broken in the meantime in master and 7.x, see https://github.com/elastic/beats/pull/26959 for example.

**Comment by ChrsMark** [comment]
I think that `master`, `7.x` and `7.14` should be compatible with 7.14 dashboards and we can fix it with a change like #24957.

The weird thing is that this PR was green at https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats/detail/PR-26919/5/pipeline, 🤔 .

In any case we need to update https://github.com/elastic/beats/blob/7f41ce048273a212ca21ff835ae63f25e1a557aa/metricbeat/docker-compose.yml#L40, no? I can prepare a PR for this if we agree.

**Comment by jsoriano** [comment]
> The weird thing is that this PR was green at https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats/detail/PR-26919/5/pipeline, thinking .

Yes, this is what I mention in https://github.com/elastic/beats/pull/26919#issuecomment-882751667, it seems it was not detected in CI because only the python tests for kubernetes were executed, what doesn't include `test_dashboards`.

> Why wasn't this detected by CI? These were the logs [there](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats/detail/PR-26919/5/pipeline/2065):
> 
> ```
> [2021-07-16T20:05:02.815Z] >> python test: Integration-kubernetes Testing
> [2021-07-16T20:05:02.815Z] Cache entry deserialization failed, entry ignored
> [2021-07-16T20:05:20.912Z] >> python test: Integration-kubernetes Testing - No tests found.
> ```

I am looking now at how we can make dashboards tests to be executed for modules too.

> In any case we need to update
> 
> https://github.com/elastic/beats/blob/7f41ce048273a212ca21ff835ae63f25e1a557aa/metricbeat/docker-compose.yml#L40

We cannot do this change till 7.14 is released. I think we need to revert the change on this dashboard, or make it work with the current tested versions. If not we are going to have broken builds till the release of 7.14.

**Pull request merged**